src/main/java/Duke.java

```diff
@@ -1,10 +1,59 @@
+import java.io.FileNotFoundException;
+
+/**
+ * Duke is a Personal Assistant Chatbot that helps a person to keep track of various things.
+ */
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Initialises a session for Duke and loads tasks, if any, from a previous session.
+     *
+     * @param filePath Location of the task list from the previous session.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (FileNotFoundException e) {
+            ui.showLoadingError(e);
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Begins a session of Duke by initialising a Duke object with the filepath to load and store the task list.
+     *
+     * @param args Unused.
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("../../../data/duke.txt").run();
+    }
+
+
+    /**
+     * Reads the program until termination.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine();
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+            } catch (DukeException e) {
+                System.err.println(e.getMessage());
```

Comment:
Printing the exceptions to stderr is a good idea, but it might be better to abstract the error printing and formatting (the "OOPS" part of the string) to the Ui class.